src/GafferModule/ScriptNodeBinding.cpp

```diff
@@ -119,78 +86,27 @@ const std::string formattedErrorContext( int lineNumber, const std::string &cont
 	);
 }
 
-// Execute the script one top level statement at a time,
-// reporting errors that occur, but otherwise continuing
-// with execution.
-bool tolerantExec( const char *pythonScript, boost::python::object globals, boost::python::object locals, const std::string &context )
+// Execute the script one line at a time, reporting errors that occur,
+// but otherwise continuing with execution.
+bool tolerantExec( const string &pythonScript, boost::python::object globals, boost::python::object locals, const std::string &context )
 {
-	// The python parsing framework uses an arena to simplify memory allocation,
-	// which is handy for us, since we're going to manipulate the AST a little.
-	std::unique_ptr<PyArena, decltype( &PyArena_Free )> arena( PyArena_New(), PyArena_Free );
-
-	// Parse the whole script, getting an abstract syntax tree for a
-	// module which would execute everything.
-	mod_ty mod = PyParser_ASTFromString(
```

Comment:
My recollection is that this `PyParser_ASTFromString()` approach was quicker than the manual line-by-line splitting approach, and script load speed is a pretty important metric for us. I think the best we can do for now is `#ifdef` for Windows so that performance doesn't suffer on Linux.

It looks like this has all changed in Python 3.9, so I'll be forced to revisit it soon enough anyway. At that point I'll do some benchmarking and try to come up with a single performant cross-platform approach.

Reply:
I added the AST code back in 4dfab183abf98c32f1643295aa58f0dfcd468e3c